codecombat/codecombat
# Acodus Chrome Ux

## Features

 - Adds functional buttons to the chrome as well as a lot of styling improvements.
 - Button handlers for the settings button and restart level button.
 - Adds a layoutChrome store module that can be queried for sound state.
 - Sound state saved in session state to preserve it through a single play-through session.
 - Cinematic and Cutscene system reacts to the sound state from Vuex.
 - Chrome buttons have tooltips.
 - Some miscellaneous layout improvements and minor bug fix in the OzariaUnitMap tests.

## How to manually test

Use a local proxy environment, using `npm run dev` and `npm run proxy` in parallel.

Log in as admin on `localhost:3000`, or your development port.

Ensure your tab isn't muted before continuing.

Navigate to the cutscene http://localhost:3000/cutscene/test-cutscene and use the toggle volume button while the cutscene is playing.

Click the map button and expect to be navigated to the teacher courses page. If you are in a courseInstance you'd be navigated to the unit map.

Navigate to  the cinematic http://localhost:3000/cinematic/1fhl1c1 and toggle the volume as the cinematic plays. 

## Comments and work to be done in future PRs

 - No major issues, but chrome volume state will likely need to be saved somewhere. Maybe in session state as a low cost solution.
 - Unit map, levels and interactives will need to respond to the sound state.
 - Likely will need to update the handlers for the settings and restart button with additional properties as needed.
